sw/device/silicon_creator/lib/drivers/otp.c

```diff
@@ -6,28 +6,39 @@
 
 #include <stddef.h>
 
-#include "sw/device/lib/base/bitfield.h"
-#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/silicon_creator/lib/base/abs_mmio.h"
 #include "sw/device/silicon_creator/lib/error.h"
 
-#include "otp_ctrl_regs.h"
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "otp_ctrl_regs.h"  // Generated.
 
-uint32_t otp_read32(const otp_t *otp, uint32_t address) {
-  return mmio_region_read32(otp->base_addr,
-                            OTP_CTRL_SW_CFG_WINDOW_REG_OFFSET + address);
+uint32_t otp_read32(uint32_t address) {
+  return abs_mmio_read32(TOP_EARLGREY_OTP_CTRL_BASE_ADDR +
+                         OTP_CTRL_SW_CFG_WINDOW_REG_OFFSET + address);
 }
 
-uint64_t otp_read64(const otp_t *otp, uint32_t address) {
+uint64_t otp_read64(uint32_t address) {
   ptrdiff_t reg_offset = OTP_CTRL_SW_CFG_WINDOW_REG_OFFSET + address;
-  uint64_t value =
-      mmio_region_read32(otp->base_addr, reg_offset + sizeof(uint32_t));
+  uint64_t value = abs_mmio_read32(TOP_EARLGREY_OTP_CTRL_BASE_ADDR +
+                                   reg_offset + sizeof(uint32_t));
   value <<= 32;
-  value |= mmio_region_read32(otp->base_addr, reg_offset);
+  value |= abs_mmio_read32(TOP_EARLGREY_OTP_CTRL_BASE_ADDR + reg_offset);
 
   return value;
 }
 
-void otp_read(const otp_t *otp, uint32_t address, void *data, size_t len) {
+rom_error_t otp_read(uint32_t address, void *data, size_t len) {
+  if (address % alignof(uint32_t) != 0 || len % alignof(uint32_t) != 0) {
```

Comment:
Consider adding TODO(#6112) to update this in the future to common alignment utility funcs.